src/borg/compress.pyx

```diff
@@ -600,6 +604,28 @@ class ObfuscateSize(CompressorBase):
             self.compressor = compressor_cls()
         return self.compressor.decompress(meta, compressed_data)  # decompress data
 
+    
+    def _padme_obfuscate(self, compr_size):
+        if compr_size <= 0:
+            return 0
+
+        E = math.floor(math.log2(compr_size))  # Get exponent (power of 2)
+        
+        # Prevent math domain error when E=0
+        if E < 1:
+            return 0  
+
```

Comment:
guess that 2nd check in 614+ can be merged into the 1st check in 609+.

Reply:
precisely because it depends you can do that merge. just don't use E, but compr_size.